src/main/java/com/amazonaws/encryptionsdk/kms/KmsMasterKey.java

```diff
@@ -51,6 +57,75 @@
     private final String id_;
     private final List<String> grantTokens_ = new ArrayList<>();
 
+    // We don't have a good method to get the original user agent string currently, so we use a bit of a hacky
+    // workaround.
+    private static final Field f_clientConfiguration;
+
+    private final String userAgent_;
+
+    static {
+        Field clientConfiguration = null;
+        try {
+            clientConfiguration = AmazonWebServiceClient.class.getDeclaredField("clientConfiguration");
+            clientConfiguration.setAccessible(true);
+        } catch (Throwable t) {
+            // Incompatible SDK change, we'll leave the Field null. Later we'll just replace the entire UA.
+        }
+
+        f_clientConfiguration = clientConfiguration;
+    }
+
+    private static String getOriginalUserAgent(AWSKMS kms) {
+        try {
+            ClientConfiguration clientConfiguration = null;
+            if (kms instanceof AWSKMSClient) {
+                clientConfiguration = (ClientConfiguration)f_clientConfiguration.get(kms);
```

Comment:
What if `f_clientConfiguration == null`?

NitPick: I know we don't do it everywhere, but I much prefer a space after a cast.